http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/request/CrtRequestAdapter.java

```diff
@@ -55,11 +64,89 @@ public static HttpRequest toCrtRequest(CrtRequestContext request) {
                                                          request.readBufferSize()));
     }
 
+    public static HttpRequest toSyncCrtRequest(CrtSyncRequestContext request) {
+
+        HttpExecuteRequest sdkExecuteRequest = request.sdkRequest();
+        SdkHttpRequest sdkRequest = sdkExecuteRequest.httpRequest();
+
+        String method = sdkRequest.method().name();
+        String encodedPath = sdkRequest.encodedPath();
+        if (encodedPath == null || encodedPath.isEmpty()) {
+            encodedPath = "/";
+        }
+
+        String encodedQueryString = sdkRequest.encodedQueryParameters()
+                                              .map(value -> "?" + value)
+                                              .orElse("");
+
+        HttpHeader[] crtHeaderArray = asArray(createSyncHttpHeaderList(sdkRequest.getUri(), sdkExecuteRequest));
+
+        ContentStreamProvider provider = sdkExecuteRequest.contentStreamProvider().isPresent() ?
+                                         sdkExecuteRequest.contentStreamProvider().get() : null;
+
+        if (provider != null) {
+            return new HttpRequest(method,
+                                   encodedPath + encodedQueryString,
+                                   crtHeaderArray,
+                                   new HttpRequestBodyStream() {
```

Comment:
Same here, can we create a separate class?

Reply:
yes